src/Stratis.Bitcoin/Consensus/ConsensusExceptions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace Stratis.Bitcoin.Consensus
+{
+    public class ConsensusException : Exception
+    {
+        protected ConsensusException() : base()
+        {
+        }
+
+        public ConsensusException(string messsage) : base(messsage)
+        {
+        }
+    }
+
+    public class MaxReorgViolationException : ConsensusException
```

Comment:
All the guys here should be commented to say when exactly they occure.

Reply:
I think we have several places where we explain max reorg violation (especially the method that throws this exception) so it will be repetition of the comments